File: CS50X/sentimental-credit/credit.py
import math

def main():
    userin = get_userinput()
    response = checksum(userin)
    if response:
        if ((userin >= 340000000000000 and userin < 350000000000000) or (userin >= 370000000000000 and userin < 380000000000000)):
                print("AMEX")
        elif (userin >= 5100000000000000 and userin < 5600000000000000):
            print("MASTERCARD")
        elif ((userin >= 4000000000000 and userin < 5000000000000) or (userin >= 4000000000000000 and userin < 5000000000000000)):
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")

def checksum(number):
    result1 = 0
    result2 = 0

    first = True

    while number != 0:
        if first:
            result1 = result1 + number % 10
            number = math.floor(number / 10)
            first = False
        else:
            next_num = (number % 10) * 2
            if next_num > 9:
                result2 = result2 + next_num % 10 + math.floor(next_num / 10)
            else:
                result2 = result2 + next_num
            number = math.floor(number / 10)
            first = True

    result = result1 + result2
    if result % 10 == 0:
        return True
    else:
        return False

def get_userinput():
    while True:
        userin = input("Number: ")
        try:
            userin = int(userin)
            if userin > 0:
                return userin
        except ValueError:
            continue

if __name__ == "__main__":
    main()

File: CS50X/credit/credit.c
#include <cs50.h>
#include <stdio.h>

long get_user_input(void);
bool checksum(long userin);

int main(void)
{
    long userin = get_user_input();
    bool response = checksum(userin);
    if (response == true)
    {
        if ((userin >= 340000000000000 && userin < 350000000000000) ||
            (userin >= 370000000000000 && userin < 380000000000000))
        {
            printf("AMEX\n");
        }
        else if (userin >= 5100000000000000 && userin < 5600000000000000)
        {
            printf("MASTERCARD\n");
        }
        else if ((userin >= 4000000000000 && userin < 5000000000000) ||
                 (userin >= 4000000000000000 && userin < 5000000000000000))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

bool checksum(long userin)
{
    int result1 = 0;
    int result2 = 0;
    bool first = true;

    while (userin != 0)
    {
        if (first)
        {
            result1 = result1 + userin % 10;
            userin = userin / 10;
            first = false;
        }
        else
        {
            int doubled_number = (userin % 10) * 2;
            if (doubled_number > 9)
            {
                result2 = result2 + doubled_number % 10 + doubled_number / 10;
            }
            else
            {
                result2 = result2 + doubled_number;
            }
            userin = userin / 10;
            first = true;
        }
    }

    int all_result = result1 + result2;
    if (all_result % 10 == 0)
    {
        return true;
    }
    else
    {
        return false;
    }
}

long get_user_input(void)
{
    while (true)
    {
        long valid_long = get_long("Number: ");
        if (valid_long > 0)
        {
            return valid_long;
        }
    }
}
